internal/provider/resource_resource_class.go

```diff
@@ -0,0 +1,247 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/humanitec/humanitec-go-autogen"
+	"github.com/humanitec/humanitec-go-autogen/client"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces
+var _ resource.Resource = &ResourceResourceClass{}
+var _ resource.ResourceWithImportState = &ResourceResourceClass{}
+
+func NewResourceResourceClass() resource.Resource {
+	return &ResourceResourceClass{}
+}
+
+// ResourceResourceClass defines the resource implementation.
+type ResourceResourceClass struct {
+	client *humanitec.Client
+	orgId  string
+}
+
+// ResourceClassModel describes the app data model.
+type ResourceClassModel struct {
+	ID           types.String `tfsdk:"id"`
+	ResourceType types.String `tfsdk:"resource_type"`
+	Description  types.String `tfsdk:"description"`
+}
+
+func (r *ResourceResourceClass) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_resource_class"
+}
+
+func (r *ResourceResourceClass) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: `DriverDefinition describes the resource driver.
+
+Resource Drivers are code that fulfils the Humanitec Resource Driver Interface.
+This interface allows for certain actions to be performed on resources such as creation and destruction.
+Humanitec provides numerous Resource Drivers “out of the box”.
+It is also possible to use 3rd party Resource Drivers or write your own.`,
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "Reflects the class string.",
```

Comment:
I wonder if this could cause issues as this way you could have multiple resources with the same id and I recall id had some special meaning in TF 🤔 

For values we use `id` as (app_id/key) and have a dedicated `key` property for the name itself https://github.com/humanitec/terraform-provider-humanitec/blob/main/internal/provider/resource_value.go#L76, might also be something to consider here.

Reply:
"The newer [Terraform Plugin Framework](https://developer.hashicorp.com/terraform/plugin/framework) is designed around modern Terraform and doesn't have all the legacy limitations of SDKv2, so for providers written using that framework the id attribute is not special and follows the same rules as for any other attribute, just like Terraform Core."
Source: https://stackoverflow.com/questions/75202696/is-it-ok-for-a-provider-to-update-resource-id-on-an-update